src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -723,6 +723,10 @@ void DismissUIAndRollbackEdits()
         }
     }
 
+    /// <remarks>
+    /// This method would create its own UIThreadOperationContext to handle cancellation.
+    /// Caller must make sure to call IUIThreadOperationContext.TakeOwnership() before calling this, otherwise it won't be able to cancel the rename operation.
+    /// </remarks>
     public void Commit(bool previewChanges = false)
```

Comment:
should these methods take a non-optional, nulalble, UIThreadOperationContext?  That way the impl can take ownership on it if present.  being non-optional means all the callsites have to try to pass their context in if available, or pass null explicitly if they really don't have a context.  it would make it easy to autdit everything that way.